ArduCopter/tuning.cpp

```diff
@@ -9,29 +10,51 @@
 //  should be called at 3.3hz
 void Copter::tuning()
 {
-    // exit immediately if tuning channel is not set
-    if (rc_tuning == nullptr) {
+    tuning(rc_tuning, g.rc_tuning_param, g2.rc_tuning_min.get(), g2.rc_tuning_max.get());
+    tuning(rc_tuning2, g2.rc_tuning2_param, g2.rc_tuning2_min.get(), g2.rc_tuning2_max.get());
+}
+
+bool Copter::being_tuned(int8_t tuning_param) const
+{
+    if (g.rc_tuning_param == tuning_param) {
+        return true;
+    }
+    if (g2.rc_tuning2_param == tuning_param) {
+        return true;
+    }
+    return false;
+}
+
+void Copter::tuning(const RC_Channel *tuning_ch, int8_t tuning_param, float tuning_min, float tuning_max)
+{
+    if (tuning_ch == nullptr) {
+        // tuning channel is not set - don't know where to take input value from
+        return;
+    }
+
+    if (tuning_param <= 0) {
+        // no parameter set for tuning
         return;
     }
-    
-    // exit immediately if the tuning function is not set or min and max are both zero
-    if ((g.rc_tuning_param <= 0) || (is_zero(g2.tuning_min.get()) && is_zero(g2.tuning_max.get()))) {
+
+    if (is_equal(tuning_min, tuning_max)) {
+        // endpoints are equal, there is no input range to tune across
         return;
     }
 
-    // exit immediately when radio failsafe is invoked or transmitter has not been turned on
-    if (!rc().has_valid_input() || rc_tuning->get_radio_in() == 0) {
```

Comment:
The zero-input-value-check has been lost here and is a little nasty if we're using the normalised-input check